internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -126,6 +137,136 @@ func (r *ClusterDeploymentsReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 	return r.updateState(ctx, clusterDeployment, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) updateClusterMetadata(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster) error {
+
+	s, err := r.getAdminPasswordSecret(ctx, cluster, c)
+	if err != nil {
+		return err
+	}
+	k, err := r.getKubeConfigSecret(ctx, cluster, c)
+	if err != nil {
+		return err
+	}
+	cluster.Spec.Installed = true
+	cluster.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
+		ClusterID: c.OpenshiftClusterID.String(),
+		InfraID:   c.ID.String(),
+		AdminPasswordSecretRef: corev1.LocalObjectReference{
+			Name: s.Name,
+		},
+		AdminKubeconfigSecretRef: corev1.LocalObjectReference{
+			Name: k.Name,
+		},
+	}
+	return r.Update(ctx, cluster)
+}
+
+func (r *ClusterDeploymentsReconciler) getAdminPasswordSecret(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster) (*corev1.Secret, error) {
+	s := &corev1.Secret{}
+	name := fmt.Sprintf(adminPasswordSecretStringTemplate, cluster.Name)
+	getErr := r.Get(ctx, types.NamespacedName{Namespace: cluster.Namespace, Name: name}, s)
+	if getErr == nil {
+		return s, nil
+	}
+	if k8serrors.IsNotFound(getErr) {
+		cred, err := r.Installer.GetCredentialsInternal(ctx, installer.GetCredentialsParams{
+			ClusterID: *c.ID,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		s = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: cluster.Namespace,
+			},
+			Data: map[string][]byte{
+				"username": []byte(cred.Username),
+				"password": []byte(cred.Password),
+			},
+		}
+
+		s.Labels = AddLabel(s.Labels, "hive.openshift.io/cluster-deployment-name", cluster.Name)
+		s.Labels = AddLabel(s.Labels, "hive.openshift.io/secret-type", "kubeadmincreds")
+
+		deploymentGVK, err := apiutil.GVKForObject(cluster, scheme.Scheme)
+		if err != nil {
+			r.Log.WithError(err).Errorf("error getting GVK for clusterdeployment")
+			return nil, err
+		}
+
+		s.OwnerReferences = []metav1.OwnerReference{{
+			APIVersion:         deploymentGVK.GroupVersion().String(),
+			Kind:               deploymentGVK.Kind,
+			Name:               cluster.ClusterName,
+			UID:                cluster.UID,
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+		}}
+
+		err = r.Create(ctx, s)
+		if err != nil {
+			return nil, err
+		}
+		return s, nil
+	}
+	return nil, getErr
+}
+
+func (r *ClusterDeploymentsReconciler) getKubeConfigSecret(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster) (*corev1.Secret, error) {
+	s := &corev1.Secret{}
+	name := fmt.Sprintf(adminKubeConfigStringTemplate, cluster.Name)
```

Comment:
Same comment as for the admin password, it may already exist with a different name int he spec.

Reply:
Name of the secret should be unique right? Why to use name and not ID? can't we have 2 clusters with the same name?

Reply:
Same as before, day 1 cluster don't have the ref to the secret.
Igal, the name is unique in the namespace.